pkg/agent/memberlist/cluster.go

```diff
@@ -257,13 +257,10 @@ func (c *Cluster) newClusterMember(node *corev1.Node) (string, error) {
 		return "", fmt.Errorf("obtain IP addresses from K8s Node failed: %v", err)
 	}
 	nodeAddr := nodeAddrs.IPv4
-	fmtStr := "%s:%d"
 	if nodeAddr == nil {
 		nodeAddr = nodeAddrs.IPv6
-		fmtStr = "[%s]:%d"
 	}
-	member := fmt.Sprintf(fmtStr, nodeAddr, c.bindPort)
-	return member, nil
+	return nodeAddr.String(), nil
```

Comment:
Doesn't memberlist expect IPv6 addrss wrapped by "[]"?

Reply:
The resolveAddr function in memberlist Join will resolve the address into an address and port, if no port is given, it will use the default. As the default BindPort has been set, so it can work.
```
	conf := memberlist.DefaultLocalConfig()
	conf.Name = c.nodeName
	conf.BindPort = c.bindPort
```
https://github.com/hashicorp/memberlist/blob/838073fef1a4e1f6cb702a57a8075304098b1c31/memberlist.go#L249

Reply:
Got it, thanks